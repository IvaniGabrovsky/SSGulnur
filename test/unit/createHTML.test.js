const { createHTML } = require('../../src/helper');

describe('createHTML', () => {
  test('.md files have to properly transform into HTML', () => {
    expect(createHTML('test/sample.md')).toMatchSnapshot();
  });

  test('.txt files have to properly transform into HTML', () => {
    expect(createHTML('test/sample.txt')).toMatchSnapshot();
  });

  test('Style should change if given with .md file', () => {
    expect(
      createHTML('test/sample.txt', 'https://cdn.jsdelivr.net/npm/water.css@2/out/water.css')
    ).toMatchSnapshot();
  });

  test('Style should change if given with .txt file', () => {
    expect(
      createHTML('test/sample.txt', 'https://cdn.jsdelivr.net/npm/water.css@2/out/water.css')
    ).toMatchSnapshot();
  });
});
