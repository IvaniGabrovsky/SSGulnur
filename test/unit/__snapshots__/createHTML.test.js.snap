// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createHTML .md files have to properly transform into HTML 1`] = `
"
    <!doctype html>
    <html lang="en">
        <head>
            <link rel="stylesheet" href="">
            <meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1">
        </head>
        <body style="background-color:#F2EECB;"> 
            <h1 id="markdowntest">Markdown test</h1><h2 id="thisfileiscreatedtotestmarkdownsyntaxstylingfeatures">This file is created to test markdown syntax styling features</h2><h1 id="h1">H1</h1><h2 id="h2">H2</h2><h3 id="h3">H3</h3><h4 id="h4">H4</h4><p>This line is to test italics in different forms such as : <em>asterisk italics</em> and <em>underscore italics</em>. </p><p>This line is to test bold in different forms such as : <strong>asterisk bold</strong> and <strong>underscore bold</strong>. </p><p>This line is to test both bold and italics combined : <strong><em>asterisk bold italics</em></strong> and <strong><em>underscore bold italics</em></strong>. </p><ol>
<li>First item 2. Second item 3. Third item </li>
</ol><ul>
<li>First item - Second item - Third item </li>
</ul><p><code>code</code> </p><hr /><p><code>javascript function() {   console.log("This is some javascript included in a markdown code block, and it will be converted to valid HTML with code syntax highlighting."); }</code> </p><p><code>html &lt;span&gt;this will remain html even after the Markdown is converted to HTML&lt;/span&gt;</code> </p><p>This line is to test link : <a href="https://github.com/gulyapulya/SSGulnur">SSGulnur</a> </p><p><img src="https://user-images.githubusercontent.com/52351598/192425336-04087ba4-0af9-4a29-a167-b84dddbd81bb.png" alt="SSGulnur logo" /> </p>
        </body>
    </html>
    "
`;

exports[`createHTML .txt files have to properly transform into HTML 1`] = `
"
    <!doctype html>
    <html lang="en">
        <head>
            <link rel="stylesheet" href="">
            <meta charset="utf-8">
            <title>Lorem Ipsum </title><meta name="viewport" content="width=device-width, initial-scale=1">
        </head>
        <body style="background-color:#F2EECB;">
            <h1>Lorem Ipsum </h1> 
            <p>What is Lorem Ipsum?  </p><p>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum. Why do we use it? It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using 'Content here, content here', making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for 'lorem ipsum' will uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like).  </p><p>Where does it come from? </p><p>Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source. Lorem Ipsum comes from sections 1.10.32 and 1.10.33 of "de Finibus Bonorum et Malorum" (The Extremes of Good and Evil) by Cicero, written in 45 BC. This book is a treatise on the theory of ethics, very popular during the Renaissance. The first line of Lorem Ipsum, "Lorem ipsum dolor sit amet..", comes from a line in section 1.10.32. </p>
        </body>
    </html>
    "
`;

exports[`createHTML Style should change if given with .md file 1`] = `
"
    <!doctype html>
    <html lang="en">
        <head>
            <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/water.css@2/out/water.css">
            <meta charset="utf-8">
            <title>Lorem Ipsum </title><meta name="viewport" content="width=device-width, initial-scale=1">
        </head>
        <body>
            <h1>Lorem Ipsum </h1> 
            <p>What is Lorem Ipsum?  </p><p>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum. Why do we use it? It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using 'Content here, content here', making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for 'lorem ipsum' will uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like).  </p><p>Where does it come from? </p><p>Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source. Lorem Ipsum comes from sections 1.10.32 and 1.10.33 of "de Finibus Bonorum et Malorum" (The Extremes of Good and Evil) by Cicero, written in 45 BC. This book is a treatise on the theory of ethics, very popular during the Renaissance. The first line of Lorem Ipsum, "Lorem ipsum dolor sit amet..", comes from a line in section 1.10.32. </p>
        </body>
    </html>
    "
`;

exports[`createHTML Style should change if given with .txt file 1`] = `
"
    <!doctype html>
    <html lang="en">
        <head>
            <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/water.css@2/out/water.css">
            <meta charset="utf-8">
            <title>Lorem Ipsum </title><meta name="viewport" content="width=device-width, initial-scale=1">
        </head>
        <body>
            <h1>Lorem Ipsum </h1> 
            <p>What is Lorem Ipsum?  </p><p>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum. Why do we use it? It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using 'Content here, content here', making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for 'lorem ipsum' will uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like).  </p><p>Where does it come from? </p><p>Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source. Lorem Ipsum comes from sections 1.10.32 and 1.10.33 of "de Finibus Bonorum et Malorum" (The Extremes of Good and Evil) by Cicero, written in 45 BC. This book is a treatise on the theory of ethics, very popular during the Renaissance. The first line of Lorem Ipsum, "Lorem ipsum dolor sit amet..", comes from a line in section 1.10.32. </p>
        </body>
    </html>
    "
`;
